salt/utils/process.py

```diff
@@ -299,13 +309,35 @@ def __init__(self,
 
         self._workers = []
 
+        # keep stats on overflows, puts, tasks_done
+        self._stats = {'overflows': 0,
+                       'puts': 0,
+                       'tasks_done': 0
+                      }
+
+        self.event = salt.utils.event.get_event(
```

Comment:
Definitely only want to do this if a configuration option is set to gather stats.